cmd/thanos/receive.go

```diff
@@ -110,38 +109,13 @@ func runReceive(
 	// initial config and mark ourselves as ready after it completed.
 	dbOpen := make(chan struct{})
 
-	// sync.Once is used to make sure we can close the channel at different execution stages(SIGTERM or when the config is loaded).
-	type closeOnce struct {
-		C     chan struct{}
-		once  sync.Once
-		Close func()
-	}
-	// Wait until the server is ready to handle reloading.
-	reloadReady := &closeOnce{
-		C: make(chan struct{}),
-	}
-	reloadReady.Close = func() {
-		reloadReady.once.Do(func() {
-			close(reloadReady.C)
-		})
-	}
-
 	level.Debug(logger).Log("msg", "setting up endpoint readiness")
 	{
 		// Initial configuration loading.
 		cancel := make(chan struct{})
 		g.Add(
 			func() error {
-				select {
-				case <-dbOpen:
-					break
-				case <-cancel:
-					reloadReady.Close()
-					return nil
-				}
-
-				reloadReady.Close()
-
+				<-dbOpen
```

Comment:
for this to not leak, we need to also close this channel in the tsdb run group no?

Reply:
hmm now that I think about it, we could really eliminate this entire run group and put the single line:

```
webHandler.Ready()
```

in the tsdb run group, no?

Reply:
in any case, we do already close it: https://github.com/improbable-eng/thanos/pull/1251/files#diff-074deab4cb04b87eb84800f5bbd8221fL176

Reply:
Not if we return right? https://github.com/improbable-eng/thanos/pull/1251/files#diff-074deab4cb04b87eb84800f5bbd8221fR144

Reply:
ah yes, you're completely right. I'll add a defer

Reply:
But yes, if there is nothing else but the readiness thing, then let's just remove the group

Reply:
ok simplified some more :) ptal when you get a change